okta/user.go

```diff
@@ -210,6 +264,31 @@ func updateAdminRolesOnUser(u string, r []interface{}, c *okta.Client) error {
 	return nil
 }
 
+// need to remove from all current groups and reassign based on terraform configs when a change is detected
+func updateGroupsOnUser(u string, g []interface{}, c *okta.Client) error {
+	groups, _, err := c.User.ListUserGroups(u, nil)
+
+	if err != nil {
+		return fmt.Errorf("[ERROR] Error Updating Groups On User: %v", err)
```

Comment:
On my current pull request I am trying to do a lot of the nice to haves, I will cut it off at some point but one thing I wanted to add is a logger/helper. You create it per CRUD action or maybe I will use dependency injection and so it will have a helper for creating user friendly errors and logging and all of that. 

Not sure why I wrote that here, just thought it would be nice haha. Less of the same log messages in all of the resources.

Reply:
Nice, that should cut down on a lot of logging code it sounds like that's currently scattered throughout the provider.  I'm going to leave this as is in this PR since it sounds like something we can tackle in a refactor PR separately.  Looking forward to that though; love getting to clean things up!